poollet/orievent/oriclassevent.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2022 OnMetal authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package orievent
+
+type Notifier interface {
+	Notify()
+}
+
+type NotifierFunc struct {
+	NotifyFunc func()
+}
+
+func (n NotifierFunc) Notify() {
+	if n.NotifyFunc != nil {
+		n.NotifyFunc()
+	}
+}
+
+type NotifierRegistration interface {
+	Remove() error
+}
+
+type ClassSource interface {
+	AddNotifier(notifier Notifier) (NotifierRegistration, error)
```

Comment:
Please follow a similar pattern as we already have for our custom 'listenable' types - i.e. 
```golang
type Notifier interface {
    AddListener(listener Listener) (ListenerRegistration, error)
    RemoveListener(reg ListenerRegistration) error
}
```